File: examples/undocumented/lua_modular/converter_isomap_modular.lua
require 'modshogun'
require 'load'

data = load_numbers('../data/fm_train_real.dat')

parameter_list = {{data}}

function converter_isomap_modular(data)
	features = modshogun.RealFeatures(data)

	converter = modshogun.Isomap()
	converter:set_target_dim(1)
	converter:apply(features)

	return features
end

if debug.getinfo(3) == nill then
	print 'Isomap'
	converter_isomap_modular(unpack(parameter_list[1]))
end

File: examples/undocumented/csharp_modular/converter_isomap_modular.cs
using System;

public class converter_classicisomap_modular {
	public static void Main() {
		modshogun.init_shogun_with_defaults();

		double[,] data = Load.load_numbers("../data/fm_train_real.dat");

		RealFeatures features = new RealFeatures(data);

		Isomap preprocessor = new Isomap();
		preprocessor.set_target_dim(1);
		preprocessor.apply(features);

		modshogun.exit_shogun();
	}
}
